### PR TITLE
[WIP] Add mariadb-galera impelentation.

### PR DESCRIPTION
**Pull request by ss7pro**
This is mariadb-galera implementation with HA.

### CODE REVIEW

File: config/mariadb-galera/templates/etc_my.cnf.j2 [hunk 1]
```diff
@@ -0,0 +1,5 @@
+[mysqld]
+datadir = /var/lib/mysql
+basedir = /usr
+[client-server]
```
Comment (DTadrzak) [inline config/mariadb-galera/templates/etc_my.cnf.j2]: add new line before section

File: config/mariadb-galera/templates/galera-seed.sh.j2 [hunk 2]
```diff
@@ -0,0 +1,78 @@
+#!/bin/sh
+
+set -e
+set -x
+
+SLEEP_TIMEOUT=5
+# we need to count seed instance here.
+MINIMUM_CLUSTER_SIZE=4
```
Comment (DTadrzak) [inline config/mariadb-galera/templates/galera-seed.sh.j2]: Could you add a wider comment?

File: config/mariadb-galera/templates/log.cnf.j2 [hunk 3]
```diff
@@ -0,0 +1,13 @@
+# ansible generated template
+# log.cnf
+[mysqld]
+slow_query_log                = off
```
Comment (DTadrzak) [inline config/mariadb-galera/templates/log.cnf.j2]: Remove redundant whitespaces.

File: config/mariadb-galera/templates/etc_my.cnf.j2 [hunk 4]
```diff
@@ -0,0 +1,5 @@
+[mysqld]
+datadir = /var/lib/mysql
```
Comment (PiotrProkop) [inline config/mariadb-galera/templates/etc_my.cnf.j2]: i would make this variable templatable

File: config/mariadb-galera/templates/peer-finder.py [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+
+   # parse to dict
+   json_acceptable_string = data.replace("'", "\"")
+   output = json.loads(json_acceptable_string)
+
+   return output
+
+
+def get_ip_addresses(output):
+   subsets = output['subsets'][0]
```
Comment (DTadrzak) [inline config/mariadb-galera/templates/peer-finder.py]: Add missing comment. Why first list element.

File: config/mariadb-galera/templates/networking.cnf.j2 [hunk 6]
```diff
@@ -0,0 +1,12 @@
+[mysqld]
+bind_address = 0.0.0.0
```
Comment (DTadrzak) [inline config/mariadb-galera/templates/networking.cnf.j2]: maybe we should use get_my_ip_address?

File: config/mariadb-galera/templates/etc_my.cnf.j2 (hunk 4 above)
Comment (DTadrzak) [inline config/mariadb-galera/templates/etc_my.cnf.j2]: I don't think so it is necessary, it is container path.

File: stackanetes-services/mariadb/mariadb-galera.yml [hunk 7]
```diff
@@ -0,0 +1,68 @@
+type: custom/mariadb-galera
```
Comment (DTadrzak) [inline stackanetes-services/mariadb/mariadb-galera.yml]: This file should replace existing mariadb.yml without this manage_all will deploy both of them.

File: stackanetes-services/mariadb/mariadb-galera.yml [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+name: mariadb-galera
```
Comment (DTadrzak) [inline stackanetes-services/mariadb/mariadb-galera.yml]: rename it to just mariadb

**Comment by PiotrProkop** [comment]
@ss7pro please close it, it is duplicate to #90.

**Pull request closed without merging**